spring-security/src/main/java/com/sap/cloud/security/spring/autoconfig/XsuaaTokenFlowAutoConfiguration.java

```diff
@@ -60,59 +58,27 @@ public XsuaaTokenFlowAutoConfiguration(XsuaaServiceConfigurations xsuaaConfigs,
 
 	@Bean
 	@Conditional(PropertyConditions.class)
-	public XsuaaTokenFlows xsuaaTokenFlows(RestOperations xsuaaRestOperations) {
+	public XsuaaTokenFlows xsuaaTokenFlows(CloseableHttpClient httpClient) {
 		logger.debug("auto-configures XsuaaTokenFlows using {} based restOperations",
 				xsuaaConfig.getClientIdentity().isCertificateBased() ? "certificate" : "client secret");
 		OAuth2ServiceEndpointsProvider endpointsProvider = new XsuaaDefaultEndpoints(xsuaaConfig);
 		ClientIdentity clientIdentity = xsuaaConfig.getClientIdentity();
-		OAuth2TokenService oAuth2TokenService = new XsuaaOAuth2TokenService(xsuaaRestOperations);
+		OAuth2TokenService oAuth2TokenService = new DefaultOAuth2TokenService(httpClient);
 		return new XsuaaTokenFlows(oAuth2TokenService, endpointsProvider, clientIdentity);
 	}
 
 	/**
-	 * Creates a {@link RestOperations} instance if the application has not defined
-	 * any.
-	 *
-	 * @return the {@link RestOperations} instance.
-	 */
-	@Bean
-	@Conditional({ OnSecretCredentialTypeCondition.class, NoClientCertificateCondition.class })
-	@ConditionalOnMissingBean
-	public RestOperations restOperations() {
-		logger.warn("In productive environment provide a well configured client secret based RestOperations bean");
-		return new RestTemplate();
-	}
-
-	/**
-	 * Creates a certificate based {@link RestOperations} instance if the
-	 * application has not defined any.
+	 * Creates a {@link CloseableHttpClient} instance configured with the
+	 * ClientIdentity provided. Conditional on missing CloseableHttpClient Bean.
 	 *
-	 * @return the {@link RestOperations} instance.
+	 * @return the {@link CloseableHttpClient} instance.
 	 */
 	@Bean
```

Comment:
Should we maybe add a bean qualifier here, e.g. @Bean(name = "xsuaaHttpClient") and @ConditionalOnMissingBean(name = "xsuaaHttpClient"). Then, users of the library who need to define further beans of type CloseableHttpClient in their application, i.e. beans not intended for communication to the XSUAA service, do not run into problems?
We use type-based injection to inject an CloseableHttpClient into the method that creates the XsuaaTokenFlows bean., so I think the **user** would be forced to **qualify** any CloseableHttpClient beans he creates to prevent them from being injected into our method. If he provides further client beans without a qualifier, they would be used by us, even if intended for a different communication channel. I think it would be good manners to leave the non-qualified bean to the main application?